app/src/main/java/com/gdsctsec/smartt/viewmodel/WeekdayActvityViewModel.kt

```diff
@@ -0,0 +1,21 @@
+package com.gdsctsec.smartt.viewmodel
+
+import android.content.Context
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.ViewModel
+import com.gdsctsec.smartt.data.TimeTable
+import com.gdsctsec.smartt.data.Weekday
+import com.gdsctsec.smartt.data.repository.LectureRepository
+
+class WeekdayActvityViewModel(val context: Context,private val weekday: String):ViewModel() {
+    private var lecturesOfTheDayLiveList: LiveData<List<TimeTable>>
+
+    init {
+        lecturesOfTheDayLiveList=LectureRepository(context,Weekday.valueOf(weekday)).getLecturesByWeekday(
```

Comment:
create a top level variable for LectureRepository()!(like you have done for lecturesOfTheDayLiveList)